docs/code-standards-policy.rst

```diff
@@ -118,36 +171,418 @@ using the debops.apt_install_ role MUST be defined as (or similar):
+configuration. For this, ``apt_preferences`` provides the following dependent
+variable:
 
 .. code-block:: yaml
 
-   - name: Install the application
-     hosts: 'application_hosts'
+   # List of :manpage:`apt_preferences(5)` pins to configure in
+   # :file:`/etc/apt/preferences.d/`.  This variable is meant to be used from a
+   # role dependency in :file:`role/meta/main.yml` or in a playbook.
+   apt_preferences__dependent_list: []
+
+The consumer role, in this case debops.nginx_ would then define an own variable
+which defines the necessary pinning information:
+
+.. code-block:: yaml
+
+   nginx__apt_preferences__dependent_list:
+     - package: 'nginx nginx-*'
+       backports: [ 'wheezy', 'precise' ]
+
+In the playbook a :ref:`soft dependency <debops_policy__ref_code_standards_soft_role_dependencies>`
+can be specified where the dependent variable is passed to the provider:
+
+.. code-block:: yaml
+
+   - name: Manage nginx webserver
+     hosts: 'debops_service_nginx'
      become: True
 
      roles:
 
        - role: debops.apt_preferences
          apt_preferences__dependent_list:
-           - '{{ application__apt_preferences__dependent_list }}'
+           - '{{ nginx__apt_preferences__dependent_list }}'
 
-       - role: application
+       - role: debops.nginx
 
-By including the configuration for the debops.apt_preferences_ role in your role's
-default variables you allow the user to change it through the Ansible inventory
+By including the configuration for the debops.apt_preferences_ role in the
+``nginx`` default variables the user to change it through the Ansible inventory
 without the need to modify any of the involved roles or the playbook.
+
+.. _debops_policy__ref_code_standards_inventory_level_scoped_variables:
+
+Inventory level scoped variables
+--------------------------------
+
+The Ansible inventory allows managing hosts, such as executing playbooks or
+scoping variables, in three different levels:
+
+- All hosts in the inventory
+
+- All hosts which are member of a common user defined group
+
+- One host individually
+
+When evaluating variable values Ansible would override variables when they are
+defined in a more specific level. This easily becomes an issue when dealing
+with lists as variable value.
+
+To mitigate this DebOps role authors SHOULD provide three different variables,
+let's call it inventory level scoped variables, for the same configuration
+property. They are meant to be defined by the user in the respective inventory
+level context. The role then has to make sure that they are merged appropriately
+so that the configuration values of all levels are respected.
+
+**Example:**
+
+The debops.users_ role allows to define user accounts which should be created
+on a machine. For accounts which should be created on every machine, the user
+can define the following variable in the global level of the inventory (e.g.
+:file:`inventory/groups_vars/all`):
```

Comment:
For this and similar paths, I would suggest to instead use something like:
```
:file:`inventory/group_vars/all/users.yml`
:file:`inventory/group_vars/group_name/users.yml`
:file:`inventory/host_vars/hostname/users.yml`
```
This allows easier management of different variable roles, where role-specific variables are in 1 file which can be easily copied/moved around if necessary. Use of the `.yml` extension enables syntax highlighting in the editor.

Of course Ansible supports multiple directory levels like this, only important thing is that directories and files shouldn't be on the same level.